core/src/main/java/hudson/tools/DownloadFromUrlInstaller.java

```diff
@@ -126,7 +127,61 @@ protected DescriptorImpl() {
             Downloadable.all().add(createDownloadable());
         }
 
-        protected Downloadable createDownloadable() {
+        /**
+         * function that creates a {@link Downloadable}.
+         * @return a downloadable object
+         */
+        public Downloadable createDownloadable() {
+            if (getId().equals("hudson.tasks.Maven.MavenInstaller") || getId().equals("hudson.tasks.Ant.AntInstaller")) {
```

Comment:
It would make more sense to have `DownloadFromUrlInstaller.DescriptorImpl` have a `reduce` method (with a static helper method that is this simple merge) and look up the extension for the `id` and if it's descriptor extends `DownloadFromUrlInstaller.DescriptorImpl` then call that's reduce method otherwise fall back to the first one wins... then just have the Maven and Ant descriptors call the static helper and it's easy for everyone